Mathlib/Order/RelIso/Set.lean

```diff
@@ -45,66 +45,61 @@ theorem range_eq (e : r ≃r s) : Set.range e = Set.univ :=
 
 end RelIso
 
-/-- `Subrel r p` is the inherited relation on a subset. -/
-def Subrel (r : α → α → Prop) (p : Set α) : p → p → Prop :=
-  (Subtype.val : p → α) ⁻¹'o r
+/-- `Subrel r p` is the inherited relation on a subset.
+
+This is defined in terms of a predicate `α → Prop` rather than a `Set α`, as otherwise the type
+signature of `Subrel r {x | p x}` is not `simp` normal, leading to hard to debug `simpNF` issues. -/
```

Comment:
I don't think we need this comment; subtypes are the more primitive object anyway, and generally we only built the Set specialization if se *also* have the subtype one.

I guess you could rephrase this as "we could also consider a `Set.Subrel r s` variant for dot notation, but this ends up interacting poorly with `simpNF`".

Reply:
I like that phrasing, I'll be copying it over.